arch/arm/core/aarch32/prep_c.c

```diff
@@ -184,6 +184,10 @@ static inline void z_arm_floating_point_init(void)
 
 extern FUNC_NORETURN void z_cstart(void);
 
+#if defined(CONFIG_EARLY_INIT)
+extern void z_early_init(void);
```

Comment:
To my understanding, the `z_` prefix is supposed to be used for Zephyr/kernel-internal functions, which this does not necessarily have to be, as noted in the commit message.

Should this be renamed to something like `sys_early_init` (or anything that does not start with `z_`)?